### PR TITLE
mac: Fix layer scale not being updated when contents is updated and window is on a new screen

### PR DESCRIPTION
**Pull request by tronical**
cc #67

### CODE REVIEW

**Comment by tronical** [comment]
I hope that saving the window is ... safe enough in terms of memory management. Otherwise this seemed like an approach simple enough :)

File: src/cg.rs [hunk 1]
```diff
@@ -27,14 +28,13 @@ impl CGImpl {
...
             layer.set_needs_display_on_bounds_change(false);
-            layer.set_contents_scale(window.backingScaleFactor());
             subview.setLayer(layer.id());
             subview.setAutoresizingMask_(NSViewWidthSizable | NSViewHeightSizable);
 
             view.addSubview_(subview); // retains subview (+1) = 2
             let _: () = msg_send![subview, release]; // releases subview (-1) = 1
         }
-        Ok(Self { layer })
+        Ok(Self { layer, window })
```
Comment (madsmtm) [inline src/cg.rs:37]: Wrt. memory management: `AppKitWindowHandle` should ideally guarantee that the window is alive, but I think I'd prefer a `let window: id = msg_send![window, retain];` before we store it, and a `let _: () = msg_send![self.window, release];` in `CGImpl`'s `Drop`.

**Review by madsmtm — CHANGES_REQUESTED** [review]
Thanks!

File: src/cg.rs (hunk 1 above)
Comment (tronical) [reply src/cg.rs:38]: Thanks, that's a good idea. Done.

**Review by madsmtm — CHANGES_REQUESTED** [review]
Nice, could you add a changelog entry as well?

**Comment by tronical** [comment]
Sure :)

**Review by madsmtm — APPROVED** [review]
Wonderful, thank you!

**Pull request merged**